runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -317,9 +317,9 @@ impl SnapshotStorageRebuilder {
             })
             .collect::<Result<HashMap<AppendVecId, Arc<AccountStorageEntry>>, std::io::Error>>()?;
 
-        let slot_entry = self.storage.entry(slot).or_default();
-        let mut storage_lock = slot_entry.write().unwrap();
-        *storage_lock = slot_stores;
+        let (id, storage) = slot_stores.drain().next().unwrap();
```

Comment:
This code is in master, and probably in 1.15. 1.15 is compatible with 1.14 snapshots for sure. It is possible, but highly unlikely that we would ever get a 1.13 snapshot from a validator with caching disabled. If so, this will not load from that snapshot... To be clear as you know caching cannot be disabled as of 1.14. Disabling the cache was a specific cli arg in 1.13. Probably not worth the trouble to make this code temporarily support snapshots created by caching disabled 1.13 validators.

Reply:
I added an assert of len 1 right above this based on other pr comments. Maybe I added one wrong? I can next look at changing this data structure up stream to make it not possible. However, the persistence format already allows Vec<> per slot, so we can't get rid of the initial possibility at some point without changing snapshot persistence, which is also not likely worth it.